### PR TITLE
feat: add codemod for `is-plain-object`

### PR DESCRIPTION
**Pull request by babs20**
Codemod for `is-plain-object` using a slightly modified replacement from the module-replacements repo. I left [an issue](https://github.com/es-tooling/module-replacements/issues/108) over there to address an edge case I encountered while creating this codemod. I am 99% confident in my variation, which matches the package's original behavior.

### CODE REVIEW

File: test/fixtures/is-plain-object/case-1/after.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+assert.strictEqual((typeof { foo: 'bar' } === "object" && ({ foo: 'bar' }.constructor === Object || { foo: 'bar' }.constructor === undefined)), true);
+assert.strictEqual((typeof {} === "object" && ({}.constructor === Object || {}.constructor === undefined)), true);
+assert.strictEqual((typeof null === "object" && null !== null && (null.constructor === Object || null.constructor === undefined)), false);
+
+assert.strictEqual((typeof 1 === "object" && 1 !== null && ((1).constructor === Object || (1).constructor === undefined)), false);
+assert.strictEqual((typeof ['foo', 'bar'] === "object" && (['foo', 'bar'].constructor === Object || ['foo', 'bar'].constructor === undefined)), false);
+assert.strictEqual((typeof [] === "object" && ([].constructor === Object || [].constructor === undefined)), false);
+
+const Foo = function () { };
+assert.strictEqual((typeof new Foo() === "object" && (new Foo().constructor === Object || new Foo().constructor === undefined)), false);
```
Comment (zloirock) [inline test/fixtures/is-plain-object/case-1/after.js:15]: `new Foo()` and other cases should be called only once -> cached since this could change state and be heavy.

Comment (babs20) [reply test/fixtures/is-plain-object/case-1/after.js]: good catch!

File: test/fixtures/is-plain-object/case-1/result.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import assert from 'assert';
+
+// Test cases
+assert.strictEqual((typeof Object.create({}) === "object" && (Object.create({}).constructor === Object || Object.create({}).constructor === undefined)), true);
+assert.strictEqual((typeof Object.create(Object.prototype) === "object" && (Object.create(Object.prototype).constructor === Object || Object.create(Object.prototype).constructor === undefined)), true);
+assert.strictEqual((typeof { foo: 'bar' } === "object" && ({ foo: 'bar' }.constructor === Object || { foo: 'bar' }.constructor === undefined)), true);
```
Comment (zloirock) [inline test/fixtures/is-plain-object/case-1/result.js:6]: Cashing should be applied, for example, for object literals too. They could have some logic like computed properties.

Comment (babs20) [reply test/fixtures/is-plain-object/case-1/result.js:6]: Ahh, I get what you mean. We should add some sort of caching into the codemod itself, right?

**Comment by babs20** [comment]
Updated this to:

1. Use an IIFE so we can use a cached version of the value we are checking.
2. Use the updated suggested replacement from the [module-replacements repo](https://github.com/es-tooling/module-replacements/blob/ca9c1399f3480abcfa679dea75c344d501e392d2/manifests/micro-utilities.json#L57-L62)

**Comment by thepassle** [comment]
This LGTM, @zloirock what do you think?

**Review by zloirock — APPROVED** [review]
👍

**Comment by thepassle** [comment]
Nice work and great collaboration guys!
